src/middlewares/permissions.js

```diff
@@ -1,15 +1,27 @@
 const PermissionsChecker = require('../services/permissions-checker');
 const httpError = require('http-errors');
 const logger = require('../services/logger');
+const ConfigStore = require('../services/config-store');
 
 const getRenderingIdFromUser = user => user.renderingId;
 
-function createCheckPermission(environmentSecret, collectionName) {
-  function checkPermission(permissionName) {
+class PermissionMiddleWareCreator {
```

Comment:
[Wording] middleware is a full word thus the camel case isn't used correctly here: `PermissionMiddlewareCreator `